app/controllers/discover.js

```diff
@@ -60,7 +60,12 @@ export default Ember.Controller.extend(Analytics, {
     queryBody: {},
     providersPassed: false,
 
-    sortByOptions: ['Relevance', 'Upload date (oldest to newest)', 'Upload date (newest to oldest)'],
+    i18n: Ember.inject.service(),
+
+    sortByOptions: Ember.computed('i18n', function() {
```

Comment:
You're on the right track here

https://github.com/jamesarosen/ember-i18n/wiki/Doc:-Translating-Text#i18nt

This needs to be `Ember.computed('i18n.locale', function`...